### PR TITLE
4.3.2 Release Updates

### PR DESCRIPTION
**Pull request by Alton09**
Upgrade Gradle plugin and target SDK to the latest versions. Needed to also update the CircleCi docker image to support API 29. Also disabled artifact desugaring by adding a property to gradle.properties files within the quickstart and quickstartKotlin modules, in response to issue #430. 

### 4.3.2 CHANGELOG Entry

Enhancement

- Upgraded the Gradle plugin used to build the SDK to 3.5.0.

Known issues

- Network handoff, and subsequent connection renegotiation is not supported for IPv6 networks [#72](https://github.com/twilio/video-quickstart-android/issues/72)
- Participant disconnect event can take up to 120 seconds to occur [#80](https://github.com/twilio/video-quickstart-android/issues/80) [#73](https://github.com/twilio/video-quickstart-android/issues/73)
- The SDK is not side-by-side compatible with other WebRTC based libraries [#340](https://github.com/twilio/video-quickstart-android/issues/340)
- Codec preferences do not function correctly in a hybrid codec Group Room with the following
codecs:
    - ISAC
    - PCMA
    - G722
    - VP9
- Unpublishing and republishing a `LocalAudioTrack` or `LocalVideoTrack` might not be seen by Participants. As a result, tracks published after a `Room.State.RECONNECTED` event might not be subscribed to by a `RemoteParticipant`.
- Using Camera2Capturer with a camera ID that does not support ImageFormat.PRIVATE capture outputs results in a runtime exception. Reference [this](https://github.com/twilio/video-quickstart-android/issues/431) issue for guidance on a temporary work around.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: quickstart/gradle.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+android.enableDexingArtifactTransform.desugaring=false
```
Comment (aaalaniz) [inline quickstart/gradle.properties]: I think we can just set this at the top level `gradle.properties`

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ aliases:
   - &build-defaults
     working_directory: *workspace
     docker:
-      - image: circleci/android:api-28-node8-alpha
+      - image: circleci/android:api-29-node
```
Comment (aaalaniz) [inline .circleci/config.yml]: Thank you for updating this, but let's use `api-28-node`

File: build.gradle [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
             'java': JavaVersion.VERSION_1_8,
-            'androidGradlePlugin': '3.3.2',
+            'androidGradlePlugin': '3.5.0',
             'googleServices': '3.2.1',
-            'compileSdk': 28,
+            'compileSdk': 29,
             'buildTools': '28.0.3',
             'minSdk': 16,
-            'targetSdk': 28,
+            'targetSdk': 29,
```
Comment (aaalaniz) [inline build.gradle]: Let's target API 28 until we confirm Android Q support.

File: quickstart/gradle.properties (hunk 1 above)
Comment (Alton09) [reply quickstart/gradle.properties]: Spent some time trying to get this to work, but with no luck. Here's the error that I am seeing when adding that line to the top level gradle.properties file.

```
FAILURE: Build failed with an exception.

* Where:

FAILURE: Build failed with an exception.

* Where:
Script '/Users/jaltonqualls/workspace/twilio/video-quickstart-android/video-quickstart-android-fork/gradle.properties' line: 1

* What went wrong:
A problem occurred evaluating script.
> Could not get unknown property 'android' for project ':exampleAdvancedCameraCapturer' of type org.gradle.api.Project.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

CONFIGURE FAILED in 0s
Could not get unknown property 'android' for project ':exampleAdvancedCameraCapturer' of type org.gradle.api.Project.
```

Comment (aaalaniz) [reply quickstart/gradle.properties]: Reference #439 for how to resolve these issues.

**Pull request merged**